### PR TITLE
f-enhancement/gh-399-handle-half-a-day-in-calendar

### PR DESCRIPTION
**Pull request by aberonja**
GH issue: #399

### CODE REVIEW

File: src/main/frontend/src/components/calendar/Day.js [hunk 1]
```diff
@@ -22,33 +23,72 @@ const pointerEventsHandler = props => {
...
+const styleHandler = (props, values) => {
+  if (!props.simplified) {
+    if (props.isToday) {
+      return values[0];
+    }
+
+    return values[1];
+  }
+
+  return values[2];
```
Comment (iscekic) [inline src/main/frontend/src/components/calendar/Day.js]: I think overriding is more readable than this (just make sure to group properties with `\n`).

File: src/main/frontend/src/components/calendar/Day.js [hunk 2]
```diff
@@ -83,22 +123,71 @@ class Day extends Component {
...
       <DayCell
-        onClick={() =>
-          this.openModal(
-            this.props.day,
-            this.props.userDay,
-            this.props.searchedEmployee
-          )
-        }
+        onClick={() => {
+          !this.props.simplified &&
```
Comment (iscekic) [inline src/main/frontend/src/components/calendar/Day.js]: Set `pointer-events: none` in simplified mode and you can remove this check.

File: src/main/frontend/src/components/calendar/Day.js [hunk 3]
```diff
@@ -83,22 +123,71 @@ class Day extends Component {
...
+        {_get(this.props, 'userDay', []).length > 0 &&
+          this.props.userDay.map(day => (
+            <RequestStatus
+              key={day.id}
+              duration={day.duration}
+              requestStatus={day.requestStatus}
+            />
+          ))}
+        {_get(this.props, 'userDay', []).length === 1 &&
+        this.props.userDay.duration !== 'FULL_DAY' ? (
```
Comment (iscekic) [inline src/main/frontend/src/components/calendar/Day.js]: Indentation?

File: src/main/frontend/src/components/calendar/Day.js [hunk 4]
```diff
@@ -83,22 +123,71 @@ class Day extends Component {
...
+          !this.props.simplified &&
+            _get(this.props, 'userDay', []).length === 0 &&
+            this.openModal(this.props.day, null, this.props.searchedEmployee);
+        }}
         className="text-center align-middle"
         searchedEmployee={this.props.searchedEmployee}
         {...this.props}
       >
-        <RequestStatus {...this.props} />
+        {_get(this.props, 'userDay', []).length > 0 &&
```
Comment (iscekic) [inline src/main/frontend/src/components/calendar/Day.js]: Extract `_get(this.props, 'userDay', []).length > 0` to variable.

Comment (iscekic) [reply src/main/frontend/src/components/calendar/Day.js]: Can we avoid this check altogether by having a `userDay` default prop?

File: src/main/frontend/src/components/calendar/Day.js [hunk 5]
```diff
@@ -83,22 +123,71 @@ class Day extends Component {
...
-        {this.props.userDay && <Image src={this.props.userDay.iconUrl} />}
+        <ImageContainer userDays={_get(this.props, 'userDay', [])}>
+          {_get(this.props, 'userDay', []).length > 0
+            ? this.props.userDay.map(day => {
+                return (
+                  <ImageBlock
+                    key={day.id}
+                    title={day.absenceName}
+                    onClick={() => {
+                      !this.props.simplified &&
```
Comment (iscekic) [inline src/main/frontend/src/components/calendar/Day.js]: Similar here as above.

File: src/main/frontend/src/components/calendar/calendarUtils.js [hunk 6]
```diff
@@ -67,10 +67,14 @@ const checkIfUserDay = dayObject => {
     return { day, publicHoliday };
   }
 
-  const userDay = personalData.find(userDay =>
+  const userDay = personalData.filter(userDay =>
     moment(userDay.date).isSame(day, 'day')
   );
 
+  if (userDay.length === 2 && userDay[0].duration === 'AFTERNOON') {
+    [userDay[0], userDay[1]] = [userDay[1], userDay[0]];
```
Comment (iscekic) [inline src/main/frontend/src/components/calendar/calendarUtils.js]:
```suggestion
    userDay.reverse();
```

File: src/main/frontend/src/components/team-calendar/ListItem.js [hunk 7]
```diff
@@ -22,7 +21,7 @@ class ListItem extends Component {
           user.days
         ).map((day, index) => {
           return (
-            <Day key={`${user.fullName}-${index}`} {...day.userDay} {...day} />
+            <Day key={`${user.fullName}-${index}`} {...day} simplified={true} />
```
Comment (iscekic) [inline src/main/frontend/src/components/team-calendar/ListItem.js]:
```suggestion
            <Day key={`${user.fullName}-${index}`} {...day} simplified />
```

File: src/main/frontend/src/components/calendar/Day.js [hunk 8]
```diff
@@ -51,10 +39,15 @@ const DayCell = styled.td`
   }
   pointer-events: ${pointerEventsHandler};
   cursor: ${ifProp('simplified', 'default', 'pointer')};
-  border: ${props =>
-    styleHandler(props, ['1px solid grey', '1px solid #c0c0c0', 'none'])};
-  border-style: ${props => styleHandler(props, ['double', 'solid', 'none'])};
   border-color: ${ifProp('isToday', 'grey')};
+  
+  border: ${props =>
+    !props.simplified && props.isToday ? '1px solid grey' : '1px solid #c0c0c0'}
```
Comment (iscekic) [inline src/main/frontend/src/components/calendar/Day.js]: No need for `!props.simplified`.

Comment (iscekic) [reply src/main/frontend/src/components/calendar/Day.js]: Same in other places as well.

File: src/main/frontend/src/components/calendar/calendarUtils.js [hunk 9]
```diff
@@ -64,15 +64,15 @@ const checkIfUserDay = dayObject => {
   const { day, publicHoliday, personalData } = dayObject;
 
   if (!personalData.length) {
-    return { day, publicHoliday };
+    return { day, userDay: [], publicHoliday };
   }
 
-  const userDay = personalData.filter(userDay =>
+  let userDay = personalData.filter(userDay =>
```
Comment (iscekic) [inline src/main/frontend/src/components/calendar/calendarUtils.js]: `userDay` can remain `const`. (Why?)

Comment (aberonja) [reply src/main/frontend/src/components/calendar/calendarUtils.js]: Because this function returns the reference of the reversed original array, right?

Comment (iscekic) [reply src/main/frontend/src/components/calendar/calendarUtils.js]: Yep. 😄

File: src/main/frontend/src/components/calendar/Day.js [hunk 10]
```diff
@@ -25,30 +26,62 @@ const pointerEventsHandler = props => {
...
+    )};
   }
+  pointer-events: ${pointerEventsHandler};
+  cursor: ${ifProp('simplified', 'default', 'pointer')};
+  border-color: ${ifProp('isToday', 'grey')};
+  
+  border: ${props => (props.isToday ? '1px solid grey' : '1px solid #c0c0c0')}
+  border: ${ifProp('simplified', 'none')}
+  
+  border: ${props => (props.isToday ? 'double' : 'solid')}
```
Comment (iscekic) [inline src/main/frontend/src/components/calendar/Day.js]:
```suggestion
  border-style: ${props => (props.isToday ? 'double' : 'solid')}
```

**Pull request merged**